Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -240,53 +459,18 @@ public struct HTTPHeaders: CustomStringConvertible {
     /// - Parameter name: The header field name whose values are to be retrieved.
     /// - Returns: A list of the values for that header field name.
     public subscript(name: String) -> [String] {
-        if let result = storage[name.lowercased()] {
-            return result.map { tuple in tuple.1 }
-        }
-        return []
+        return storage.findAll(name: name).map { storage.string(idx: $0) }
     }
 
     /// Serializes this HTTP header block to bytes suitable for writing to the wire.
     ///
     /// - Parameter buffer: A buffer to write the serialized bytes into. Will increment
     ///     the writer index of this buffer.
     func write(buffer: inout ByteBuffer) {
-        for (key, values) in storage {
-            if key != "set-cookie" {
-                writeListHeaderValues(buffer: &buffer, key: key, values: values)
-            } else {
-                writeSequentialHeaderValues(buffer: &buffer, key: key, values: values)
-            }
-        }
-        buffer.write(staticString: crlf)
-    }
-
-    /// Used for most HTTP headers, which can be represented as a single line joined by commas.
-    private func writeListHeaderValues(buffer: inout ByteBuffer, key: String, values: [(String, String)]) {
-        buffer.write(string: key)
-        buffer.write(staticString: headerSeparator)
-
-        var writerIndex = buffer.writerIndex
-        for (_, value) in values {
-            buffer.write(string: value)
-            writerIndex = buffer.writerIndex
-            buffer.write(staticString: ",")
-        }
-        // Discard last ,
-        buffer.moveWriterIndex(to: writerIndex)
+        self.storage.write(into: &buffer, continous: self.continous)
         buffer.write(staticString: crlf)
     }
 
-    /// Used for HTTP headers that cannot be joined with commas, e.g. set-cookie.
```

Comment:
Does this correctly handle set-cookie now?

Reply:
If our tests included something for it yes... I will double-check